### PR TITLE
Adform adapter added

### PR DESCRIPTION
**Pull request by braizhas**

### CODE REVIEW

**Comment by mkendall07** [comment]
Hi @braizhas 

Thanks for submitting your adapter. I have reviewed it and have one question about the sizes. Your request doesn't seem to input the ad size that you would expect. When I plug in your test ad params, I get a 100x100 size back into a 300x250 spot. Is there a way to pass through dimensions? Currently we don't validate the ad size fits the spot back so it will still display the ad. Thanks!

**Comment by braizhas** [comment]
Hi, 100x100 is a default size for example placement (mid=74042) in Adform system. If there is a need to override default settings, size could be passed via "w" and "h" params. See example below:
{
  bidder: 'adform',
  params: {
    mid: 74042,
    test: 1,
    w: 300,
    h: 250
  }
}

**Comment by protonate** [comment]
Hi @braizhas, actually the placement size info is already set on the adUnit object [here](https://github.com/prebid/Prebid.js/blob/master/integrationExamples/gpt/pbjs_example_gpt.html#L63), so you can remove the height and width properties from the AdForm params and pull them in from the adUnit properties.

**Comment by braizhas** [comment]
Hello @protonate, at the moment there is no possibility to pass multiple dimensions with request to Adform SSP, but placements associated with *mid* parameter could be set up to target multiple creative sizes. So, only one size could be pulled from adUnit object. Passing size with request prevents ability to win this bid for creatives with other size which are also related to *mid*. My suggestion is to check if creative size fits adUnit size after bids are back. I will commit related changes soon.

**Comment by protonate** [comment]
Ok, I see how `verifySize` is preventing a 100x100 response. Is it possible to see an example where the size matches the placement?

Two minor notes -- please remove the `./dist/*` build artifacts and use triple equality `var a === b;`, thanks.

File: dist/prebid.js [hunk 1]
```diff
@@ -129,6 +131,157 @@ exports.createNew = function(bidderCode){
...
+        return encode64(url.join(''));
+    }
+
+    function handleCallback(bids) {
+        return function handleResponse(adItems) {
+            var bidObject, bidder = 'adform', adItem, bid;
+            for(var i = 0, l = adItems.length; i < l; i++){
+                adItem = adItems[i];
+                bid = bids[i];
+                if (adItem && adItem.response == 'banner' &&
```
Comment (protonate) [inline dist/prebid.js]: use triple equality `===`

File: dist/prebid.js [hunk 2]
```diff
@@ -129,6 +131,157 @@ exports.createNew = function(bidderCode){
...
+                    bidObject = bidfactory.createBid(2);
+                    bidObject.bidderCode = bidder;
+                    bidmanager.addBidResponse(bid.placementCode, bidObject);
+                }
+            }
+        };
+
+        function verifySize(adItem, validSizes) {
+            for (var j = 0, k = validSizes.length; j < k; j++) {
+                if (adItem.width == validSizes[j][0] &&
```
Comment (protonate) [inline dist/prebid.js]: use triple equality `===`

File: dist/prebid.js [hunk 3]
```diff
@@ -129,6 +131,157 @@ exports.createNew = function(bidderCode){
...
+                    adItem.height == validSizes[j][1]) {
```
Comment (protonate) [inline dist/prebid.js]: use triple equality `===`

File: integrationExamples/gpt/pbjs_example_adform_gpt.html [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+    */
+    function initAdserver() {
+        if (pbjs.initAdserverSet) return;
+
+        //load GPT library here
+        (function() {
+            var gads = document.createElement('script');
+            gads.async = true;
+            gads.type = 'text/javascript';
+            var useSSL = 'https:' == document.location.protocol;
```
Comment (protonate) [inline integrationExamples/gpt/pbjs_example_adform_gpt.html]: use triple equality `===`

File: src/adapters/adform.js [hunk 5]
```diff
@@ -0,0 +1,149 @@
...
+        return encode64(url.join(''));
+    }
+
+    function handleCallback(bids) {
+        return function handleResponse(adItems) {
+            var bidObject, bidder = 'adform', adItem, bid;
+            for(var i = 0, l = adItems.length; i < l; i++){
+                adItem = adItems[i];
+                bid = bids[i];
+                if (adItem && adItem.response == 'banner' &&
```
Comment (protonate) [inline src/adapters/adform.js]: use triple equality `===`

File: src/adapters/adform.js [hunk 6]
```diff
@@ -0,0 +1,149 @@
...
+                    bidObject = bidfactory.createBid(2);
+                    bidObject.bidderCode = bidder;
+                    bidmanager.addBidResponse(bid.placementCode, bidObject);
+                }
+            }
+        };
+
+        function verifySize(adItem, validSizes) {
+            for (var j = 0, k = validSizes.length; j < k; j++) {
+                if (adItem.width == validSizes[j][0] &&
```
Comment (protonate) [inline src/adapters/adform.js]: use triple equality `===`

File: src/adapters/adform.js [hunk 7]
```diff
@@ -0,0 +1,149 @@
...
+                    adItem.height == validSizes[j][1]) {
```
Comment (protonate) [inline src/adapters/adform.js]: use triple equality `===`

**Comment by protonate** [comment]
There's a few more triple equality fixes I flagged in code comments, then please squash commits and we can get this merged. Thanks!

**Comment by braizhas** [comment]
@protonate, Squashed.

**Comment by protonate** [comment]
@braizhas I mean to please flatten into a single commit or this PR, for example like [this](http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html).

**Comment by mkendall07** [comment]
@braizhas 
Could you remove the `/dist/*` resources from this PR? Thanks.

**Comment by protonate** [comment]
Commits were flattened and merged manually, closing this PR, thanks for contributing the Adform adapter.

**Comment by braizhas** [comment]
Thanks guys! Sorry for my ignorance, I am on holidays now.

**Pull request closed without merging**